Weather forecast line chart and display weather card

rutujapadgilwar opened the pull request:
## Description
This pull request adds the functionality for displaying a weather forecast of 5 days using a line chart and a box card. The line chart visualizes the low and high-temperature trends, and the box card shows the low and high temperatures, weather description, and icon for each day. Additionally, different colors are assigned based on the weather forecast to enhance the visual representation.

## Changes Made
- Implemented a line chart component to display the temperature trends for 5 days.
- Created a box card component to show the low and high temperatures for each day.
- Integrated weather forecast data to populate the line chart and box card.
- Defined color mappings for different weather conditions, enhancing the visual representation.
- Created everything as media responsive. 

## Screenshots
<img width="734" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/106363277/2fcb6a3f-6ed7-4388-84da-1ac5b52a91a0">

Please let me know your feedback

rutujapadgilwar committed:
Weather forecast line chart and display weather card

rutujapadgilwar committed:
Add default color for weather discription box card

rutujapadgilwar committed:
Update the activity icons for unique key value

schoure98 commented on park-dashboard/src/Components/ParkActivities/ParkActivities.js:
```diff
@@ -5,70 +5,69 @@ import {
+
 function ParkActivities(item) {
   const classes = useStyles();
   const parkData = LoadParkData(item);
```
Thanks for adding this LoadParkData component in common for /parks endpoint. Its very useful while fetching data from local storage.

schoure98 commented on park-dashboard/src/Components/WeatherForecast/WeatherForecast.js:
```diff
@@ -0,0 +1,311 @@
+      const containerWidth = weatherForecastRef.current.offsetWidth;
+      const totalWidth = dailyForecast.length * 200;
+      if (totalWidth > containerWidth) {
+        weatherForecastRef.current.classList.add("scrollable");
```
Nice work with adding and removing scroll. Can this be all time visible scrollbar? Since its horizontal scroll, I think it will be better to indicate the scrollbar to user.

schoure98 commented on park-dashboard/src/Components/WeatherForecast/WeatherForecast.js:
```diff
@@ -0,0 +1,311 @@
+
+  const groupDataByDay = (data) => {
+    const groupedData = {};
+    data.forEach((item) => {
```
If you use the reduce function instead of foreach, the mutation of groupedData can be avoided. 
And I believe that will also make the code more readable.

schoure98 commented on park-dashboard/src/Components/WeatherForecast/WeatherForecast.js:
```diff
@@ -0,0 +1,311 @@
+
+  const calculateDailyData = (groupedData) => {
+    const dailyData = [];
+    for (const date in groupedData) {
```
I strongly suggest use of Object.entries on grouped data and then using map function for key values. Instead of manually creating an empty array and pushing values to it these ES6 functions might help to ease the functioning as a better practice. Although the code is working correctly and fulfilling the purpose.

rutujapadgilwar replied:
Thank you for your feedback, I have added the Object.entries on grouped data now.

schoure98 commented:
<img width="750" alt="Screenshot 2023-05-24 at 11 10 48 PM" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/114700046/55c4b9cf-ddca-4e66-ab81-8c1ed7689ab4">

The color for light rain is not same as the expected screenshot. And the temperature is overwritten on min and max temp, can you add space in between that?

rutujapadgilwar committed:
add object.entries on grouped data

rutujapadgilwar committed:
Add suggestedMax temperature for y axis so the temperature value can't overwrite

rutujapadgilwar commented:
> <img alt="Screenshot 2023-05-24 at 11 10 48 PM" width="750" src="https://user-images.githubusercontent.com/114700046/240818713-55c4b9cf-ddca-4e66-ab81-8c1ed7689ab4.png">
> The color for light rain is not same as the expected screenshot. And the temperature is overwritten on min and max temp, can you add space in between that?

Updated the code now the color is displaying properly, also I add the suggestMex temperature for the y-axis so now it's not overwritten the max and min temperature values.
<img width="693" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/106363277/f39d63fc-80cf-4b95-92ec-df6149eab42f">

juhiaugustine commented on park-dashboard/src/Components/WeatherForecast/WeatherForecast.js:
```diff
@@ -0,0 +1,312 @@
+import { GiRaining } from "react-icons/gi";
+import { Line } from "react-chartjs-2";
+import ChartDataLabels from "chartjs-plugin-datalabels";
+import Chart from "chart.js/auto";
```
It seems like this import is unused. Can you remove this?

juhiaugustine commented on park-dashboard/src/Components/WeatherForecast/WeatherForecast.js:
```diff
@@ -0,0 +1,312 @@
+import React, { useState, useEffect, useRef } from "react";
+import moment from "moment";
+import "./WeatherForecast.css";
+import { WiCloudy, WiRain, WiSnow } from "react-icons/wi"; // Import weather icons from react-icons library
```
Can you consolidate both the imports together as they are from the same library? Also WiCloud seems to be not used. Can you remove it if it's not being used?

rutujapadgilwar replied:
Consolidated the import sentences.

juhiaugustine commented:
Great work on the weather component. Looks really good!
Just dropped some minor comments...

rutujapadgilwar committed:
Added responsive media functionality to ensure optimal viewing experience across different devices and remove unwanted lines

rutujapadgilwar commented:
> Great work on the weather component. Looks really good! Just dropped some minor comments...

Thanks, juhi, As per your suggestion I have removed the unwanted lines.
Moreover, I have made the necessary code updates, including adding media responsiveness functionality. This includes implementing appropriate CSS for the ParkInfoPage and Navbar sections, Additionally, I've made the necessary updates to the "Alters" component view on the parkInfoPage to ensure its adaptability to various screen sizes.
